### PR TITLE
fix servo bug restoring state and starting servo detached

### LINKED ISSUE #964: Servo starts moving on startup
**Operating environment/Installation (Hass.io/Docker/pip/etc.):**

Hassio on Raspbian Buster , Using ESPHome (1.14.3) as an add on

**ESP (ESP32/ESP8266, Board/Sonoff):** 

Wemos - ESP8266
**Affected component:**

https://esphome.io/components/servo.html 

**Description of problem:**
I am trying to run a servo using ESPHome, while the servo is working fine and I am able to control it well, it starts rotating as soon as the ESP boots up. I dont know why is it behaving so. I have even detached the servo on_boot but that does not have any affect on this behavior. 
I can be sure that there is nothing wrong with the servo or the Wemos as I am able to run the servo well using normal Arduino code. I have been running the same using my own code for long and wanted to switch to ESPHome for the curtain project I am working on. 
In the code below I am using one button to flip the direction of the servo and another to stop the servo. Both work well except that on start up the motor starts spinning at a slow speed, I have to flip the stop button to stop that. Things are fine thereafter.

**Problem-relevant YAML-configuration entries:**
```yaml
substitutions:
  static_ip: XXX.XXX.XX.XX
  status_gpio: '2'
  btn_up: 'D2'
  btn_stop: 'D1'

esphome:
  name: curtain_test
  platform: ESP8266
  board: d1_mini
  on_boot:
    lambda: |-
      {
        id(curtain_test_servo).detach();
        ESP_LOGI("main","detached servo:");
      }

#    - servo.detach: curtain_test_servo

wifi:
  ssid: !secret primary_ssid
  password: !secret primary_ssid_pswd
  manual_ip:
    static_ip: $static_ip
    gateway: !secret default_gateway
    subnet: !secret subnet_mask

# Enable logging
logger:

# Enable Home Assistant API
api:

ota:
#  password: !secret ota_pswd
  
<<: !include include/wifiinfo.yaml

servo:
  - id: curtain_test_servo
    output: pwm_output
    restore: false



# Example output platform
# On ESP32, use ledc output
output:
  - platform: esp8266_pwm
    id: pwm_output
    pin: D0
    frequency: 50 Hz

binary_sensor:
  - platform: gpio
    pin:
      number: $btn_up
      mode: INPUT_PULLUP
    name: "Up"
    id: "curtain_up"
    on_press:
      then:
      - servo.write:
          id: curtain_test_servo
          level: -100.0%
    on_release:
      then:
      - servo.write:
          id: curtain_test_servo
          level: 100.0%

  - platform: gpio
    pin:
      number: $btn_stop
      mode: INPUT_PULLUP
    name: "Stop"
    id: "curtain_stop"
    on_state:
      then:
        lambda: |-
            id(curtain_test_servo).detach();

status_led:
  pin:
    number: $status_gpio
    inverted: yes
    
```

**Logs (if applicable):**

```
PASTE DEBUG LOG HERE

```

**Additional information and things you've tried:**
The only difference I see between my Arduino code and ESPHome config is that in Arduino code I detach the servo when I stop it and attach it when I have to move it. In ESPHome one can only detach it and attaching seems to be automatic. Not sure if that's the reason. 
For reference snippets of my Ardunio code:

```
Servo myservo;

void prepCurtainMove()
{
    servoRunning = true;
    //save the current state so that if it crashes before stopping cleanly, we know that it crashed in between running
    state curr_state = {.steps = currentStepPosition, .isRunning = servoRunning, .maxSteps = curtainMaxSteps};
    saveCurrentState(curr_state);
    if(!myservo.attached())
      myservo.attach(MOTOR_PIN);
    lastEncoderTime = millis();//initialize the last encoder time as we're starting to move now.
}

void curtainStop()
{
    if(myservo.attached())
      myservo.detach();
    servoRunning = false;
    delay(1000);//wait for a second for the motor to stop completely
    updateEncoder();//update the final stopped encoder position 
    state curr_state = {.steps = currentStepPosition, .isRunning = servoRunning, .maxSteps = curtainMaxSteps};
    saveCurrentState(curr_state);
    setLED(0);//turn the LED OFF
    DPRINTLN("Curtain stopped at position:");DPRINTLN(String(currentStepPosition));
}

void curtainUp()
{
  if(!btnTop)
  {
    if((currentStepPosition > 0) || crashRecovery)// TO DO do direction check here
    {
      curtainDirectionUp = true;
      prepCurtainMove();
      myservo.write(180);
      lastEncoderTime = millis();//initialize the last encoder time as we're starting to move now.
      DPRINTLN("Curtain is moving up");
    }
    else
      DPRINTLN("Curtain up command ignored as curtain is at the top");
  }
  else
      DPRINTLN("Curtain up command ignored as curtain is at the extreme top");
}
  
void curtainDn()
{
  if((currentStepPosition < curtainMaxSteps) || crashRecovery) // TO DO do direction check here
  {
    curtainDirectionUp = false;
    prepCurtainMove();
    myservo.write(0);
    lastEncoderTime = millis();//initialize the last encoder time as we're starting to move now.
    DPRINTLN("Curtain is moving down");
  }
  else
    DPRINTLN("Curtain down command ignored as curtain is at the bottom");
}

```

### PR DESCRIPTION
**Pull request by glmnet**
## Description:

Fixes as small bug which caused the restore option to actually not work.
Also servo by default now starts detached (I think this was the intended behavior), so I'm flagging this as breaking change (before the servo will start as centered and you would have to override this value with some nasty on_boot)

**Related issue (if applicable):** fixes https://github.com/esphome/issues/issues/964

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by glmnet** [comment]
@brandond do you know why this has "Some checkes haven't completed yet"?

**Comment by Skaronator** [comment]
@glmnet Looks like travis had some issues reaching the github api. Either rebase you commit and push again or restart the build here (if you have the permissions)

https://travis-ci.org/github/esphome/esphome/builds/670387533

**Comment by glmnet** [comment]
@Skaronator restarting the build seems to have fixed it, thanks.
